[Reland][ONNX] Move all torch.onnx.export related tests to test/onnx

titaiwangms opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #87292

Moving torch.onnx.export related tests to test/onnx integrates ONNX tests to the same CI machine, so the testing environment can be better managed.

Fixes https://github.com/pytorch/pytorch/issues/87320

titaiwangms committed:
[ONNX] Move tall orch.onnx.export related tests to test/onnx

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Move tall orch.onnx.export related tests to test/onnx"


Moving torch.onnx.export related tests to test/onnx integrates ONNX tests to the same CI machine, and the testing environment can be better managed.

[ghstack-poisoned]

AllenTiTaiWang commented on test/onnx/test_pytorch_onnx_no_runtime.py:
```diff
@@ -427,7 +428,7 @@ def forward(self, x):
             MyClip(),
             torch.randn(3, 4, requires_grad=True),
-            custom_ops=[common_utils.custom_op("aten::clamp", bad_clamp, 9)],
+            custom_ops=[common_utils.custom_op("aten::clamp", bad_clamp, GLOBALS.export_onnx_opset_version)],
```
This test was broken for a while. I wonder why CI didn't catch this.

AllenTiTaiWang replied:
Found the current CI only has Caffe2 included version, so the tests with `skipIfCaffe2` is skipped. Those tests are all aten_fallback related.

thiagocrepaldi replied:
We need an issue for that. ONNX Runtime heavily depends on BUILD_CAFFE2=0 builds to properly work and PyTorch CI is not guarding those scenarios

AllenTiTaiWang commented on test/onnx/test_export_modes.py:
```diff
@@ -66,10 +61,10 @@ def foo(a):
             return (a, a)
         f = io.BytesIO()
         x = torch.ones(3)
-        torch.onnx._export(foo, (x,), f)
```
in `torch.onnx._export`, it has a parameter called export_type that is not exposed in `torch.onnx.export`, but we have its tests right above blocks. I am thinking if we should delete those as it's not exposed anyway?

thiagocrepaldi replied:
That should be fine. We don't need to export to all formats. ONNX uses protobuf anyway

titaiwangms committed:
Update on "[ONNX] Move tall orch.onnx.export related tests to test/onnx"


Moving torch.onnx.export related tests to test/onnx integrates ONNX tests to the same CI machine, so the testing environment can be better managed.

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Move tall orch.onnx.export related tests to test/onnx"


Moving torch.onnx.export related tests to test/onnx integrates ONNX tests to the same CI machine, so the testing environment can be better managed.

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Move tall orch.onnx.export related tests to test/onnx"


Moving torch.onnx.export related tests to test/onnx integrates ONNX tests to the same CI machine, so the testing environment can be better managed.

[ghstack-poisoned]

AllenTiTaiWang commented:
@malfet please take a look. It's moving tests cases.

BowenBao commented on test/quantization/eager/test_quantize_eager_ptq.py:
```diff
@@ -1444,26 +1443,6 @@ def forward(self, indices, offsets, linear_in):
         self.assertTrue('DynamicQuantizedLinear' in str(q_model))
 
 class TestQuantizeEagerONNXExport(JitTestCase):
```
this class has onnx name in there somehow..

malfet replied:
Yes, but other teams are using it, we should coordinate the name change with them

BowenBao commented:
@kit1980 I think this likely is out of onnx merge rules. Please take a look.

titaiwangms committed:
Update on "[ONNX] Move all torch.onnx.export related tests to test/onnx"


Moving torch.onnx.export related tests to test/onnx integrates ONNX tests to the same CI machine, so the testing environment can be better managed.

Fixes https://github.com/pytorch/pytorch/issues/87320

[ghstack-poisoned]

AllenTiTaiWang commented:
@pytorchbot merge -g

AllenTiTaiWang commented:
@pytorchbot merge -g

AllenTiTaiWang commented:
@pytorchbot merge -g

weiwangmeta commented:
@pytorchbot revert -m "breaking internal test relating to quantization eager tests, see test/quantization/eager/test_quantize_eager_ptq.py test_lower_graph_linear and test_lower_graph_conv2d " -c ghfirst

AllenTiTaiWang commented:
> What should users do if they encounter errors (AttributeError: 'TestQuantizeEagerONNXExport' object has no attribute '_test_lower_graph_impl' ) due to the following deletion? Would this be breaking whenever "self._test_lower_graph_impl(model, data)" when self is TestQuantizeEagerONNXExport?
> 
> <img alt="image" width="548" src="https://user-images.githubusercontent.com/109318740/199066860-8035d390-be1a-41a7-81ba-0a314550421b.png">

I can't see internal test failing. The goal of here is to move torch.onnx.export tests into test/onnx, so that we can have our API tested on the same environment. Is there a way you can do that in these specific tests? Or anything I can do?

weiwangmeta commented:
> test/quantization/eager/test_quantize_eager_ptq.py test_lower_graph_linear and test_lower_graph_conv2d

Can you please test the above two cases? If they pass, then internal should be able to pass. You may need to remove the two skips (FBGEMM and CAFFE2) to trigger the two cases. 
Details below: 
https://github.com/pytorch/pytorch/blob/master/test/quantization/eager/test_quantize_eager_ptq.py#L1468-L1482

titaiwangms committed:
Update on "[ONNX] Move all torch.onnx.export related tests to test/onnx"


Moving torch.onnx.export related tests to test/onnx integrates ONNX tests to the same CI machine, so the testing environment can be better managed.

Fixes https://github.com/pytorch/pytorch/issues/87320

[ghstack-poisoned]

titaiwangms committed:
Update on "[Reland][ONNX] Move all torch.onnx.export related tests to test/onnx"


Moving torch.onnx.export related tests to test/onnx integrates ONNX tests to the same CI machine, so the testing environment can be better managed.

Fixes https://github.com/pytorch/pytorch/issues/87320

[ghstack-poisoned]

malfet commented:
Thank you for the follow up, looks good to me

AllenTiTaiWang commented:
> Thank you for the follow up, looks good to me (though ghstack for a single PR feels a bit excessive)

Thanks for the help.

AllenTiTaiWang commented:
@pytorchbot merge -g
